adafruit_tinylora/adafruit_tinylora.py

```diff
@@ -0,0 +1,405 @@
+        time.sleep(0.01)
+        # switch interrupt to txdone
+        self._write_u8(0x40, 0x40)
+        # check for multi-channel configuration
+        if self._channel is None:
+            self._tx_random = randint(0, 7)
+            self._rfm_lsb = self._frequencies[self._tx_random][2]
+            self._rfm_mid = self._frequencies[self._tx_random][1]
+            self._rfm_msb = self._frequencies[self._tx_random][0]
+        # write to RFM channel registers...
+        self._write_u8(_REG_FRF_MSB, self._rfm_msb)
+        self._write_u8(_REG_FRF_MID, self._rfm_mid)
+        self._write_u8(_REG_FRF_LSB, self._rfm_lsb)
+        # set RFM datarate
+        self._write_u8(_REG_FEI_LSB, self._sf)
+        self._write_u8(_REG_FEI_MSB, self._bw)
+        self._write_u8(_REG_MODEM_CONFIG, self._modemcfg)
+        # set RegPayloadLength
+        self._write_u8(_REG_PAYLOAD_LENGTH, packet_length)
+        # initalize FIFO pointer to base address for TX
+        self._write_u8(_REG_FIFO_POINTER, _REG_FIFO_BASE_ADDR)
+        # fill the FIFO buffer with the LoRa payload
+        k = 0  # ptr
+        i = 0
+        while i < packet_length:
+            self._write_u8(0x00, lora_packet[k])
+            i += 1
+            k += 1
+        # switch RFM to TX operating mode
+        self._write_u8(_REG_OPERATING_MODE, _MODE_TX)
+        print('Sending packet')
+        # wait for TxDone IRQ, poll for timeout.
+        start = time.monotonic()
+        timed_out = False
+        while not timed_out and not self._irq.value:
+            if(time.monotonic() - start) >= timeout:
+                timed_out = True
+        print('Packet Sent!')
+        # switch RFM to sleep operating mode
+        self._write_u8(_REG_OPERATING_MODE, _MODE_SLEEP)
+        if timed_out:
+            raise RuntimeError('Timeout during packet send')
+
+    def set_datarate(self, datarate):
+        """Sets the RFM Datarate
+        :param datarate: Bandwidth and Frequency Plan
+        """
+        if datarate == 'SF7BW125':
+            self._sf = 0x74
+            self._bw = 0x72
+            self._modemcfg = 0x04
+        elif datarate == 'SF7BW250':
+            self._sf = 0x74
+            self._bw = 0x82
+            self._modemcfg = 0x04
+        elif datarate == 'SF8BW125':
+            self._sf = 0x84
+            self._bw = 0x72
+            self._modemcfg = 0x04
+        elif datarate == 'SF9BW125':
+            self._sf = 0x94
+            self._bw = 0x72
+            self._modemcfg = 0x04
+        elif datarate == 'SF10BW125':
+            self._sf = 0xA4
+            self._bw = 0x72
+            self._modemcfg = 0x04
+        elif datarate == 'SF11BW125':
+            self._sf = 0xB4
+            self._bw = 0x72
+            self._modemcfg = 0x0C
+        elif datarate == 'SF12BW125':
+            self._sf = 0xC4
+            self._bw = 0x72
+            self._modemcfg = 0x0C
+        else:
+            raise TypeError("Invalid Datarate.")
```

Comment:
Could store all this in a dict of tuples:
```python
data_rates = {'SF7BW125':(0x74, 0x72, 0x04), 'SF7BW250':(0x74, 0x82, 0x04), **etc** }
```
and here, just do:
```python
self._sf, self._bw, self._modemcfg = data_rates[datarate]
```

Reply:
Added and tested, throwing an error if requested `datarate` is not in the dict.